src/caja-navigation-window-menus.c

```diff
@@ -735,7 +759,11 @@ action_show_hide_search_callback (GtkAction *action,
         g_object_unref (location);
 
         caja_navigation_window_hide_search (window);
+    /* needed to exclude everything inside the `else` */
+    G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
     }
+    G_GNUC_END_IGNORE_DEPRECATIONS;
+    /*end deprecation check*/
```

Comment:
```
    G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
    }
    G_GNUC_END_IGNORE_DEPRECATIONS;
```

I don't understand this change, it seems to ignore nothing, I think it isn't needed.

If you like just to ignore the warning of the line `if (gtk_toggle_action_get_active (GTK_TOGGLE_ACTION (action)))`

I suggest just to use a gboolean variable, I think it is more clear and readable:

```
gboolean variable;
...
...
...
G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
variable = gtk_toggle_action_get_active (GTK_TOGGLE_ACTION (action));
G_GNUC_END_IGNORE_DEPRECATIONS;

if (variable)
```

Reply:
This bracket is the ending from the else condidition. Without this caja-searchbar don't work.

Reply:
Is your suggestion tested?

Reply:
I didn't tested but it must work, you can believe me :)

Reply:
Done, and it works , thx.